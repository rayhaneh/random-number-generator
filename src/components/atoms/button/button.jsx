import * as React from 'react';

export const Button = (props) => {
  const {
    text,
  } = props;

  return (
    <button 
      className={'f3 bg-blue white pa2'}
      type='submit'
    >
      {text}
    </button>
  );
}
