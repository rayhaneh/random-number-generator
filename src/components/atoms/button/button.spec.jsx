import * as React from 'react';
import {
  configure,
  shallow,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import { Button } from './button';

let props;
let comp;

describe('Button Component', () => {
  beforeEach(() => {
    props = {
      text: 'Submit'
    }
  });

  it('renders properly', () => {
    comp = shallow(<Button {...props} />);
    expect(comp).toMatchSnapshot();
  });
});
