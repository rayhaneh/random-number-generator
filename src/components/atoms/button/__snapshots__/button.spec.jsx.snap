// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="Submit"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Submit",
      "className": "f3 bg-blue white pa2",
      "type": "submit",
    },
    "ref": null,
    "rendered": "Submit",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Submit",
        "className": "f3 bg-blue white pa2",
        "type": "submit",
      },
      "ref": null,
      "rendered": "Submit",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
